src/main/java/seedu/address/logic/commands/SearchCommand.java

```diff
@@ -29,17 +34,46 @@ public class SearchCommand extends Command {
             + "Example 3: " + COMMAND_WORD + " n/ alice t/ friend\n"
             + "Example 4: " + COMMAND_WORD + " t/ friend n/ alice";
 
+    public static final String MESSAGE_NO_FOUND_GROUP = "There is no such group!";
+
     private final Predicate<Person> predicate;
+    private final String groupName;
 
-    public SearchCommand(Predicate<Person> predicate) {
+    /**
+     * Constructs a {@code SearchCommand} to filter the person list using the specified predicate and group name.
+     *
+     * @param predicate The predicate that filters the list of persons based on name and/or tag criteria.
+     * @param groupName The name of the group to further filter the persons.
+     *                  If empty or {@code null}, no group filtering is applied.
+     */
+    public SearchCommand(Predicate<Person> predicate, String groupName) {
         this.predicate = predicate;
+        this.groupName = groupName;
     }
 
 
     @Override
-    public CommandResult execute(Model model) {
+    public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        model.updateFilteredPersonList(predicate);
+
+        Predicate<Person> finalPredicate = predicate;
+
+        // If a group name is provided, perform a group lookup and 1bine it with the existing predicate
+        if (groupName != null && !groupName.isEmpty()) {
+            // Fetch the GroupList
+            GroupList groupList = model.getAddressBook().getGroupList();
+            try {
+                // Retrieve the group by name
+                Group group = groupList.get(groupName);
+                GroupContainsKeywordsPredicate groupPredicate = new GroupContainsKeywordsPredicate(group);
+
+                // Combine predicates
+                finalPredicate = finalPredicate == null ? groupPredicate : finalPredicate.and(groupPredicate);
```

Comment:
I think this one could also be expanded a bit to improve readability. Let me know what you think!

Reply:
If we were to expand, it would something like below.
```
if (finalPredicate == null) {
    finalPredicate = groupPredicate;
} else {
    finalPredicate = finalPredicate.and(groupPredicate);
}
```
I am thinking the original one would be better as it is more concise? Perhaps i could include parentheses to improve readability. 

```
finalPredicate = (finalPredicate == null) ? groupPredicate : finalPredicate.and(groupPredicate);
```

Would this be better?